spinalcordtoolbox/vertebrae/core.py

```diff
@@ -156,28 +186,31 @@ def vertebral_detection(fname, fname_seg, contrast, param, init_disc, verbose=1,
             break
         # find next disc
         # N.B. Do not search for C1/C2 disc (because poorly visible), use template distance instead
-        if current_disc != 1:
-            current_z = compute_corr_3d(data, data_template, x=xc, xshift=0, xsize=param.size_RL,
+        if current_disc != 1 and current_disc != 2:
+            current_z = compute_corr_3d(data_hm, im_lab.data, x=xc, xshift=0, xsize=param.size_RL,
```

Comment:
I think `im_lab.data` and `data_template` are the same thing? They both refer to `Image(fname_template)`, I think.

Reply:
yes they are.

Reply:
In that case, perhaps `im_lab` could be removed to stick with `data_template`

Reply:
Yes right !